sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_fileshare_storage_helper.py

```diff
@@ -318,3 +322,11 @@ def recursive_download(
             target=ErrorTarget.ARTIFACT,
             error_category=ErrorCategory.SYSTEM_ERROR,
         )
+
+    def generate_sas(self, **kwargs):
+        """Not supported by file share."""
+        raise NotImplementedError("SAS generation not supported by FileShare storage.")
+
+    def update_metadata(self, *args) -> None:
+        """Not supported by file share"""
+        raise NotImplementedError("Metadata not supported by FileShare storage.")
```

Comment:
I've started building a unified API between the different Storage-type wrapper objects.
This would allow us to remove all the Storage-type validation throughout the code and just have the wrapper implement  the required features or raise like here where it's not supported.
(These `NotImeplementedError` could be swapped out for your validation error for example.)